File: MovieRatingPredictor.Model/Models/Statistics/Statistics.cs
using MovieRatingPredictor.Model.Models.Dtos;

namespace MovieRatingPredictor.Model.Models.Statistics;

public class Statistics
{
    public CategoryStatistics CategoryStatistics { get; set; } = new();
    public GrossStatistics GrossStatistics { get; set; } = new();
    public RuntimeStatistics RuntimeStatistics { get; set; } = new();
    public YearStatistics YearStatistics { get; set; } = new();
    public ImdbRatingStatistics ImdbRatingStatistics { get; set; } = new();

    public IEnumerable<UserStatisticDto> MaxRatedUserStatistic { get; set; }
    public IEnumerable<UserStatisticDto> MinRatedUserStatistic { get; set; }
}

File: MovieRatingPredictor.WebClient/ClientApp/src/app/core/models/Statistics.ts
import { UserStatisticDto } from './dto/UserStatisticsDto';

export interface YearStatistics {
  yearCountDictionary: { [key: string]: number };
}

export interface CategoryStatistics {
  crimeFlagCount: number;
  horrorFlagCount: number;
  comedyFlagCount: number;
  romanceFlagCount: number;
  musicFlagCount: number;
  adventureFlagCount: number;
  mysteryFlagCount: number;
  warFlagCount: number;
  westernFlagCount: number;
  biographyFlagCount: number;
  historyFlagCount: number;
  thrillerFlagCount: number;
  scifiFlagCount: number;
  actionFlagCount: number;
  dramaFlagCount: number;
}

export interface RuntimeStatistics {
  min: number;
  max: number;
}

export interface ImdbRatingStatistics {
  max: number;
  min: number;
}

export interface GrossStatistics {
  max: number;
  min: number;
}

export interface Statistics {
  categoryStatistics: CategoryStatistics;
  grossStatistics: GrossStatistics;
  runtimeStatistics: RuntimeStatistics;
  yearStatistics: YearStatistics;
  imdbRatingStatistics: ImdbRatingStatistics;
  maxRatedUserStatistic: UserStatisticDto[];
  minRatedUserStatistic: UserStatisticDto[];
}
